api/envoy/config/route/v3/route_components.proto

```diff
@@ -1944,3 +1956,15 @@ message InternalRedirectPolicy {
   // x-forwarded-proto. The default is false.
   bool allow_cross_scheme_redirect = 4;
 }
+
+// A simple wrapper for an HTTP filter config. This is intended to be used as a wrapper for the
+// map value in
+// :ref:`VirtualHost.typed_per_filter_config<envoy_api_field_config.route.v3.VirtualHost.typed_per_filter_config>`,
+// :ref:`Route.typed_per_filter_config<envoy_api_field_config.route.v3.Route.typed_per_filter_config>`,
+// or :ref:`WeightedCluster.ClusterWeight.typed_per_filter_config<envoy_api_field_config.route.v3.WeightedCluster.ClusterWeight.typed_per_filter_config>`
+// to indicate that the filter is optional. In this case, if the client does not support the filter,
+// it may ignore the map entry rather than rejecting the config.
+// [#not-implemented-hide:]
+message OptionalFilterConfig {
+  google.protobuf.Any config = 1;
+}
```

Comment:
Is this wrapper needed?  Setting is_optional=true ought to disable parsing of config for filters that are optional and not build into the binary.  It should be possible to ignore this config without wrapping it in an extra layer of messages.

Reply:
The problem is that at the time that we receive the RouteConfiguration from RDS, we don't necessarily know which HCM config it's going to be matched up with, because there could be a new LDS response waiting to be swapped in, so we can't depend on the is_optional=true that was set there.  See my comment in #12274 for details.

Reply:
Yeah, I think this is unavoidable given the nature of xDS resource independence.